src/bin/ion/commands/dump.rs

```diff
@@ -63,10 +63,7 @@ pub fn run(_command_name: &str, matches: &ArgMatches<'static>) -> Result<()> {
         }
     } else {
         let input: StdinLock = stdin().lock();
-        // TODO: This `BufReader` is only required because StdInLock doesn't yet implement
-        //       ToIonDataSource. It's probably unnecessary overhead.
-        let buf_reader = BufReader::new(input);
-        let mut reader = ReaderBuilder::new().build(buf_reader)?;
```

Comment:
🗺️ This problem was [fixed in Rust v1.61](https://blog.rust-lang.org/2022/05/19/Rust-1.61.0.html#static-handles-for-locked-stdio).